### PR TITLE
InfiltrateForPlayerUpgrades

### PR DESCRIPTION
**Pull request by phrohdoh**
Add InfiltrateForPlayerUpgrades so the infiltrating player can unlock tech, or do anything else upgrade-related.

Will be used in RA2.

### CODE REVIEW

File: OpenRA.Mods.Common/OpenRA.Mods.Common.csproj [hunk 1]
```diff
@@ -715,4 +716,7 @@ cd "$(SolutionDir)"</PostBuildEvent>
   <Target Name="AfterBuild">
   </Target>
   -->
+  <ItemGroup>
```
Comment (pchote) [inline OpenRA.Mods.Common/OpenRA.Mods.Common.csproj]: I think our policy is to not merge these groups (which aren't actually necessary).

**Comment by pchote** [comment]
This should in principle obsolete the existing `InfiltrateForSupportPower` trait.

File: OpenRA.Mods.Common/OpenRA.Mods.Common.csproj (hunk 1 above)
Comment (penev92) [inline OpenRA.Mods.Common/OpenRA.Mods.Common.csproj]: Yeah, next time I modify the project VS will want to remove this.

Comment (Phrohdoh) [inline OpenRA.Mods.Common/OpenRA.Mods.Common.csproj]: `InfiltrateForPlayerUpgrades` is in `Infiltration`.

**Comment by Phrohdoh** [comment]
Removed unnecessary `ItemGroup`.

File: OpenRA.Mods.Common/Infiltration/InfiltrateForPlayerUpgrades.cs [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+#endregion
+
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	[Desc("Grants upgrades to the infiltrating actor's owner.")]
+	public class InfiltrateForPlayerUpgradesInfo : ITraitInfo
+	{
+		[Desc("Upgrades to grant when this actor is infiltrated."), FieldLoader.Require]
```
Comment (Mailaender) [inline OpenRA.Mods.Common/Infiltration/InfiltrateForPlayerUpgrades.cs]: This is missing `[UpgradeGrantedReference]`. See https://github.com/OpenRA/OpenRA/pull/8321.

Comment (Phrohdoh) [inline OpenRA.Mods.Common/Infiltration/InfiltrateForPlayerUpgrades.cs]: Good point, fixing.

**Comment by Phrohdoh** [comment]
Added the missing `[UpgradeGrantedReference]` attribute.

**Comment by Phrohdoh** [comment]
Good game, Github.
I'll open a new PR.

**Pull request closed without merging**